### ISSUE TITLE
propagation-stackdriver causes "Error: socket hang up "

### ISSUE THREAD

**Issue by chrisdostert** [census-instrumentation/opencensus-node#110]
# Repro
- Wire up tracing per below code snippet
- Start application & make a request

# Expected
traces show up in stackdriver

# Actual
no traces show up in stackdriver and the application crashes w/ the error snippet below a minute or so after receiving a few requests

of note, if I remove propogation traces DO show up in stackdriver so it's something related to the use of `@opencensus/propogation-stackdriver`

# code snippet
```javascript
const tracing = require('@opencensus/nodejs')
const stackdriverKeyFile = require('/stackdriverKeyFile.json')
const propagation = require('@opencensus/propagation-stackdriver')
const exporter = require('@opencensus/exporter-stackdriver')

tracing.start({
  exporter: new exporter.StackdriverTraceExporter({
    projectId: stackdriverKeyFile.project_id
  }),
  logLevel: 4,
  propagation: propagation.v1
})
```

# error snippet
```shell
Error: socket hang up
    at createHangUpError (_http_client.js:335:15)
    at TLSSocket.socketOnEnd (_http_client.js:428:23)
    at TLSSocket.emit (events.js:185:15)
    at endReadableNT (_stream_readable.js:1101:12)
    at process._tickCallback (internal/process/next_tick.js:114:19)
Emitted 'error' event at:
    at TLSSocket.socketOnEnd (_http_client.js:428:9)
    at TLSSocket.emit (events.js:185:15)
    at endReadableNT (_stream_readable.js:1101:12)
    at process._tickCallback (internal/process/next_tick.js:114:19)
```

**Comment by chrisdostert** [comment]
Interesting, I just tried using `require('@opencensus/propagation-b3')` and it has the same result so it appears the issue does not lie with '@opencensus/propagation-stackdriver'

**Comment by antonj** [comment]
I also get this error when using `@opencensus/exporter-stackdriver` but not when using the `@opencensus/exporter-jaeger`so my guess is that is has something to do with the stackdriver exporter.

**Comment by frankcs** [comment]
Same situation here with `require('@opencensus/propagation-b3')`. 
The `Error: socket hang up` is happening with node `v8.12.0`. 
When using with node `v10.12.0` the issue dissapears and traces show up in stackdriver.

**Comment by isaikevych** [comment]
blocked by https://github.com/census-instrumentation/opencensus-node/issues/158

**Comment by isaikevych** [comment]
@chrisdostert, @antonj, Could you please provide stackdriver version or try one more time with latest master?
I found another issue https://github.com/census-instrumentation/opencensus-node/issues/158 which was fixed, and I cant reproduce current in master, node v8.12.0, I checked `opencensus/exporter-stackdriver` and `@opencensus/propagation-b3`
